06x-new-kafka-roller.md

```diff
@@ -0,0 +1,195 @@
+| <b>preferredLeaderElectionDelay</b>   | 10 seconds             | No             | Delay right after a node restart and before triggering partition leader election so that just-rolled broker is leading all the preferred replicas. This is to avoid situations where leaders moving to a newly started node that does not yet have established networking to some outside networks, e.g. through load balancers.
+
+### Future improvement
+
+In the future, we can optimize Cruise Control's `BrokerSetAwareGoal` to make the batch rolling more efficient when user enabled Cruise Control in their clusters. This goal operates at the level of broker sets, which may correspond to physical or logical boundaries like racks, data centres, availability zones or custom logical groupings defined by administrators. It ensures that replicas of a partition are not assigned to the same broker set by spreading them across sets as evenly as possible. The new roller could then restart brokers in the same set in parallel while maintaining availability of partitions. This goal can be used for both rack aware and non rack aware clusters, comparing to `RackAwareGoal` which relies on Kafka's built-in `broker.rack` property.
+
+This solution still has the limitation mentioned in the `Rejected Alternatives` section, that we can't be certain that other tooling hasn't reassigned some replicas since the last rebalance. In this case, the proposed algorithm can be used to check that brokers in the same set have no common partitions. This can be discussed further in the future.
+
+
+### Feature Gate
+
+The switch from the old roller to the new roller should be controlled by a new feature gate called `UseNewKafkaRoller`.
+With this feature gate disabled, the operator will continue using the old KafkaRoller.
+With it enabled, the new roller will be used.
+The following table shows the expected graduation of the feature gate:
+
+| Phase | Strimzi versions       | Default state                                          |
+|:------|:-----------------------|:-------------------------------------------------------|
+| Alpha | 0.46, ?                | Disabled by default                                    |
+| Beta  | ? , ?                  | Enabled by default                                     |
+| GA    | ?, ?                   | Enabled by default (without possibility to disable it) |
+
+Currently, I'm not sure what would be the feature gate timeline as it is subject to change based on the actual progress.
+However, each phase probably will take at least 2 releases. 
+
+## Affected/not affected projects
+
+This proposal affects only
+the [`strimzi-kafka-operator`](https://github.com/strimzi/strimzi-kafka-operator).
+
+## Compatibility
+
+The new KafkaRoller introduced by this proposal will used only for KRaft based clusters. 
+This proposal should have no impact on any existing Kafka clusters deployed with ZooKeeper. 
```

Comment:
Because Strimzi doesn't support Zookeeper anymore, are these two lines needed? Just asking out of curiosity, as I know that the proposal was written in the moment when we supported both.